From 81073a178c198a1b7f08964300905cbec595c14e Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@kernel.org>
Date: Thu, 31 Oct 2024 14:54:21 +0100
Subject: [PATCH] linux: fix derive_psk_digest OpenSSL 1.1 version

The reordering of the variables broke derive_psk_digest OpenSSL 1.1
version.

Fixes: 4cc727d19221 ("linux: reorder variable declarations")

Signed-off-by: Daniel Wagner <wagi@kernel.org>
---
 src/nvme/linux.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/nvme/linux.c b/src/nvme/linux.c
index e74fac20ba18..0d48792431d0 100644
--- a/src/nvme/linux.c
+++ b/src/nvme/linux.c
@@ -881,14 +881,14 @@ int nvme_gen_dhchap_key(char *hostnqn, enum nvme_hmac_alg hmac,
 }
 
 static int derive_psk_digest(const char *hostnqn, const char *subsysnqn,
-			     int version, int hmac,
+			     int version, int cipher,
 			     unsigned char *retained, size_t key_len,
 			     char *digest, size_t digest_len)
 {
 	static const char hmac_seed[] = "NVMe-over-Fabrics";
-	const EVP_MD *md = select_hmac(hmac, &hmac_len);
 	_cleanup_hmac_ctx_ HMAC_CTX *hmac_ctx = NULL;
 	_cleanup_free_ unsigned char *psk_ctx = NULL;
+	const EVP_MD *md;
 	size_t hmac_len;
 	size_t len;
 
-- 
2.47.0

